pkg/validation/test.go

```diff
@@ -515,18 +536,26 @@ func validateLiteralTestStep(context context, stage testStage, step api.LiteralT
 					if !context.releases.Has(releaseName) {
 						ret = append(ret, fmt.Errorf("%s.from: unknown imagestream '%s'", context.fieldRoot, imageParts[0]))
 					}
-
 				}
 			}
 		}
 	}
 	if len(step.Commands) == 0 {
 		ret = append(ret, fmt.Errorf("%s: `commands` is required", context.fieldRoot))
+	} else {
+		ret = append(ret, validateCommands(step)...)
+	}
+
+	if step.BestEffort != nil && *step.BestEffort && step.Timeout == nil {
+		ret = append(ret, fmt.Errorf("test %s contains best_effort without timeout", step.As))
 	}
+
 	ret = append(ret, validateResourceRequirements(context.fieldRoot+".resources", step.Resources)...)
 	ret = append(ret, validateCredentials(context.fieldRoot, step.Credentials)...)
-	if err := validateParameters(&context, step.Environment); err != nil {
-		ret = append(ret, err)
+	if context.env != nil {
+		if err := validateParameters(&context, step.Environment); err != nil {
+			ret = append(ret, err)
+		}
 	}
 	ret = append(ret, validateDependencies(context.fieldRoot, step.Dependencies)...)
 	ret = append(ret, validateLeases(context.forField(".leases"), step.Leases)...)
```

Comment:
I missed this one: we'll also want to do a similar check in there to skip the duplicate test.